integration-tests/integration_test.go

```diff
@@ -525,6 +553,76 @@ func (s *RepeatedNetworkFlowTestSuite) TestRepeatedNetworkFlow() {
 	fmt.Printf("ClientDetails from test: %s %s\n", s.clientContainer, s.clientIP)
 }
 
+// Launches nginx container
+// Launches gRPC server in insecure mode
+// Launches collector
+// Note it is important to launch the nginx container before collector, which is the opposite of 
+// other tests. The purpose is that we want ConnScraper to see the nginx endpoint and we do not want
+// NetworkSignalHandler to see the nginx endpoint.
+func (s *ConnScraperTestSuite) SetupSuite() {
+
+	s.metrics = map[string]float64{}
+	s.executor = NewExecutor()
+	s.StartContainerStats()
+	s.collector = NewCollectorManager(s.executor, s.T().Name())
+
+	s.collector.Env["COLLECTOR_CONFIG"] = `{"logLevel":"debug","turnOffScrape":` + strconv.FormatBool(s.turnOffScrape) + `,"scrapeInterval":2}`
+
+	s.launchNginx()
+
+	err := s.collector.Setup()
+	s.Require().NoError(err)
+
+	err = s.collector.Launch()
+	s.Require().NoError(err)
+
+	time.Sleep(10 * time.Second)
+
+
+	s.logLines = s.GetLogLines("grpc-server")
+
+	err = s.collector.TearDown()
+	s.Require().NoError(err)
+
+	s.db, err = s.collector.BoltDB()
+	s.Require().NoError(err)
+}
+
+func (s *ConnScraperTestSuite) launchNginx() {
+	image := "nginx:1.14-alpine"
+
+	err := s.executor.PullImage(image)
+	s.Require().NoError(err)
+
+	// invokes default nginx
+	containerID, err := s.launchContainer("nginx", image)
+	s.Require().NoError(err)
+	s.serverContainer = containerShortID(containerID)
+
+	time.Sleep(10 * time.Second)
+}
+
+func (s *ConnScraperTestSuite) TearDownSuite() {
+	s.cleanupContainer([]string{"nginx", "collector"})
+	stats := s.GetContainerStats()
+	s.PrintContainerStats(stats)
+	s.WritePerfResults("ConnScraper", stats, s.metrics)
+}
+
+func (s *ConnScraperTestSuite) TestConnScraper() {
```

Comment:
This test only checks for the container id. Can you also validate all of the other fields?

Reply:
Done. Though I had difficulty with the network address and ended up leaving that out.